android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -275,6 +275,28 @@ public void isAirPlaneMode(Promise p) {
     p.resolve(isAirPlaneMode);
   }
 
+  @ReactMethod
+  public void isAutoDateAndTime(Promise p) {
+    String name;
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR1) {
+      name = Settings.System.AUTO_TIME;
+    } else {
+      name = Settings.Global.AUTO_TIME;
+    }
+    p.resolve(Settings.System.getInt(this.reactContext.getContentResolver(), name, 0) != 0);
+  }
+
+  @ReactMethod
+  public void isAutoTimeZone(Promise p) {
+    String name;
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR1) {
+      name = Settings.System.AUTO_TIME_ZONE;
+    } else {
+      name = Settings.Global.AUTO_TIME_ZONE;
+    }
+    p.resolve(Settings.Global.getInt(this.reactContext.getContentResolver(), name, 0) != 0);
+  }
+
```

Comment:
This is really really close to being perfect, however, if you get the name from one of Settings.System or Settings.Global, you also need to resolve it from that source. There is an example of what I mean in the airplane mode check just above this one - the whole content resolution is switched on the Build.VERSION_CODE, not just the name resolution: https://github.com/rebeccahughes/react-native-device-info/blob/0de338f2aded9bea71195b9f4565aa1f9505a00f/android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java#L268

Reply:
@mikehardy Yea, I just didn't notice that part of it. Made this changes after suggestion from @machour. Now I feel I could just revert the commit.

Reply:
If I recall I think the suggestion from @machour was to protect the API on VERSION.SDK_INT < Build.VERSION_CODES.JELLEY_BEAN_MR1. That part is still vital, as react-native works on those versions, but the API changed during them.

The part I mean is that the *entire call* (including the content resolution part, not just the location of the name) needs protection by API version

Which is to say I don't think a revert is called for, a more thorough API protection / usage is called for

Reply:
@mikehardy Yea, I made a mistake in my initial commit due to which @machour tried to refactor it and I again failed to notice the change. Any thanks for marking it, resolved it now.